src/Aspire.Hosting/Publishing/ManifestPublishingContext.cs

```diff
@@ -269,6 +269,31 @@ public void WritePortBindingEnvironmentVariables(IResource resource)
         }
     }
 
+    internal void WriteDockerBuildArgs(IEnumerable<DockerBuildArg>? buildArgs)
+    {
+        var args = buildArgs?.ToArray() ?? [];
+        if (args is { Length: > 0 })
+        {
+            Writer.WriteStartObject("buildArgs");
+
+            for (var i = 0; i < args.Length; i++)
+            {
+                var buildArg = args[i];
+
+                var valueString = buildArg.Value switch
+                {
+                    string stringValue => stringValue,
+                    IManifestExpressionProvider manifestExpression => manifestExpression.ValueExpression,
+                    _ => null
```

Comment:
Curious why we don't just `ToString()` in the catch all case?

Reply:
Because the `null` case. If it falls through, and the `value` is `null`, that would be `null.ToString()`. Instead, we just want `null`.

Reply:
Done